src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.libgpiod.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Device.Gpio;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    private const string LibgpiodLibrary = "System.Device.Gpio.Native";
+
+    /// <summary>
+    ///     Create a new gpiochip iterator.
+    /// </summary>
+    /// <returns>Pointer to a new chip iterator object or NULL if an error occurred.</returns>
+   [DllImport(LibgpiodLibrary, SetLastError = true)]
+   internal static extern SafeChipIteratorHandle GetChipIterator();
+
+   /// <summary>
+   ///     Release all resources allocated for the gpiochip iterator and close the most recently opened gpiochip(if any).
+   /// </summary>
+   /// <param name="iter">The gpiochip iterator object</param>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern void FreeChipIterator(IntPtr iter);
+
+   /// <summary>
+   ///     Release all resources allocated for the gpiochip iterator but don't close the most recently opened gpiochip (if any).
+   /// </summary>
+   /// <param name="iter">The gpiochip iterator object</param>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern void FreeChipIteratorNoCloseCurrentChip(SafeChipIteratorHandle iter);
+
+   /// <summary>
+   ///     Get the next gpiochip handle.
+   /// </summary>
+   /// <param name="iter">The gpiochip iterator object</param>
+   [DllImport(LibgpiodLibrary, SetLastError = true)]
+   internal static extern SafeChipHandle GetNextChipFromChipIterator(SafeChipIteratorHandle iter);
+
+   /// <summary>
+   ///     Close a GPIO chip handle and release all allocated resources.
+   /// </summary>
+   /// <param name="chip">The GPIO chip pointer</param>
+   [DllImport(LibgpiodLibrary)]
+   internal static extern void CloseChip(IntPtr chip);
```

Comment:
SafeChipHandle instead of IntPtr?

Reply:
SafeHandle is the wrapper of IntPtr, closing/releasing methods should be pass IntPtr

Reply:
@joperezr yes, this is called inside of the handle's dispose so you cannot use safe handle (i've hit issues with that in the past for other classes - basically you don't know what state the object is in during the tear-down so the internal IntPtr might already be zero)